manifests.go

```diff
@@ -0,0 +1,66 @@
+package distribution
+
+import (
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/digest"
+	"github.com/docker/distribution/reference"
+)
+
+// Manifest represents a registry object specifying a target and a set of constituents
+type Manifest interface {
+	// Target returns a descriptor for the configuration of this manifest.  This
+	// may return a nil descriptor if none exists for this manifest.
+	Target() Descriptor
+
+	// Constituents returns a list of objects which make up this manifest.
+	// The dependencies are strictly ordered from base to head. A constituent
+	// is anything which can be represented by a distribution.Descriptor
+	Constituents() []Descriptor
```

Comment:
I am also not a fan of the name, but if it remains unchanged this will be the last I say about it :smile: